sharktank/sharktank/layers/base.py

```diff
@@ -33,12 +41,82 @@ def _set_recursively_submodules_default_trace_tensor_key_prefix(
         )
 
 
-class BaseLayer(nn.Module):
+def get_model_type_id(model_type: type["BaseLayer"]) -> str:
+    """Get a string representation of the model type."""
+    return f"{model_type.__module__}.{model_type.__name__}"
+
+
+def create_model(config: PathLike | ModelConfig | Mapping[str, Any], /) -> "BaseLayer":
+    """
+    Create model from a configuration.
+    Example
+
+    config.json:
+    ```
+    {
```

Comment:
We may want to version these configs somehow. We should at least think through forwards/backwards compatibility a bit (could be "only supported within the same version").

Reply:
I added versioning.